docs-src/docs/guides/walkthrough.md

```diff
@@ -0,0 +1,317 @@
+```
+
+In the next phase of deployment, if we execute the `service get` command again, we will see **(1)** in the **Scale** column; this indicates that 1 copy of the service is transitioning to an active state:
+```
+SERVICE ID    SERVICE NAME   ENVIRONMENT  LOADBALANCER  DEPLOYMENTS       SCALE
+guestbo9364b  guestbook-svc  demo-env     guestbook-lb  guestbook-dep:1*  0/1 (1)
+```
+
+In the final phase of deployment, we should see that the service has been fully scaled up to the desired count. If we execute the `service get` command again, you will see the following output:
+
+```
+SERVICE ID    SERVICE NAME   ENVIRONMENT  LOADBALANCER  DEPLOYMENTS       SCALE
+guestbo9364b  guestbook-svc  demo-env     guestbook-lb  guestbook-dep:1   1/1
+```
+
+
+---
+
+### Get the Application's URL
+
+Once the service has been completely deployed, we can obtain the URL for the application and launch it in a browser.
+
+At the command prompt, execute the following:
+
+`l0 loadbalancer get demo-env:guestbook-lb`
+
+This will give us the following output:
+
+```
+LOADBALANCER ID  LOADBALANCER NAME  ENVIRONMENT  SERVICE        PORTS       PUBLIC  URL
+guestbodb65a     guestbook-lb       demo-env     guestbook-svc  80:80/HTTP  true    <url>
+```
+
+Copy the value shown in the **URL** column and paste it into a web browser. The guestbook application will appear (once the service has completely finished deploying).
+
+
+---
+
+### Cleanup
+
+If you're finished with the example and don't want to continue with this walkthrough, you can instruct Layer0 to delete the environment and terminate the application and the Layer0 the application was using.
+
+`l0 environment delete demo-env`
+
+However, if you intend to continue through [Deployment 2](#deployment-2-guestbook-redis), you will want to keep the resources you made in this section.
+
+
+---
+
+## 1b: Deploy with Terraform
+
+Instead of using the Layer0 CLI directly, you can instead use the Layer0 Provider for Terraform, and deploy using Terraform. 
+
+
+---
+
+### Part 1: Download the Configuration Files
+
+- [Guestbook.Dockerrun.aws.json](https://github.com/quintilesims/layer0-examples/blob/guides-overhaul/iterative-walkthrough/guestbook/Guestbook.Dockerrun.aws.json)
+- [terraform.tfvars](https://github.com/quintilesims/layer0-examples/blob/guides-overhaul/iterative-walkthrough/terraform/terraform.tfvars)
+- [layer0.tf](https://github.com/quintilesims/layer0-examples/blob/guides-overhaul/iterative-walkthrough/terraform/deployment-1/layer0.tf) for Deployment 1
+
+
+---
+
+### Part 2: Terraform Apply
+
+Run `terraform apply` to begin the process. Terraform will prompt you for configuration values that it does not have.
+
+To begin deploying the application, run the following command:
+
+`terraform apply`
+
+_To avoid entering these values manually each time you run terraform, you can set the terraform variables by editing the `terraform.tfvars` file._
+
+```
+var.endpoint
```

Comment:
Link to terraform configuration options: https://www.terraform.io/docs/configuration/variables.html

They have the options of:
* Entering the vars at the command promp
* Entering the vars into `terraform.tfvars`
* Setting environment variables, e.g. `TF_VAR_endpoint` and `TF_VAR_token`